it/uk/gov/hmrc/eusubsidycompliancefrontend/util/EscTestWireMock.scala

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.eusubsidycompliancefrontend.util
+
+import com.github.tomakehurst.wiremock.WireMockServer
+import com.github.tomakehurst.wiremock.client.{ResponseDefinitionBuilder, WireMock}
+import com.github.tomakehurst.wiremock.matching.RequestPatternBuilder
+import com.github.tomakehurst.wiremock.stubbing.{ServeEvent, StubMapping}
+import play.api.libs.json.Writes
+import uk.gov.hmrc.eusubsidycompliancefrontend.models.types.{EORI, UndertakingRef}
+import uk.gov.hmrc.eusubsidycompliancefrontend.models.{ApproveEmailAsVerifiedByEoriRequest, ApproveEmailByVerificationIdRequest, BusinessEntity, StartEmailVerificationRequest, SubsidyRetrieve, SubsidyUpdate, Undertaking, UndertakingCreate}
+
+import java.time.LocalDate
+import scala.jdk.CollectionConverters.CollectionHasAsScala
+
+/**
+  * Ring fence this into a variable as loose methods in tests from traits just turns in to a brute force memory
+  * exercise that eats velocity when writing tests.
+  */
+class EscTestWireMock(wireMockServer: WireMockServer, baseUrl: String = "/eu-subsidy-compliance") {
+
+  private implicit class MappingBuilderOps(responseDefinitionBuilder: ResponseDefinitionBuilder) {
+    def withJsonHeader: ResponseDefinitionBuilder = {
+      responseDefinitionBuilder.withHeader("content-type", "application/json")
+
+      responseDefinitionBuilder
+    }
+  }
+
+  //language=json - connector tests don't care about response structure but the more detailed integration tests do
+  val defaultResponse: String =
+    """
+      |{
+      |  "key" : "value"
+      |}
+      |""".stripMargin
+
+  def serveEvents: List[ServeEvent] = wireMockServer.getAllServeEvents.asScala.toList
+
+  object createUndertaking {
+    private val url = s"$baseUrl/undertaking"
+    def stubExpected(statusCode: Int): Unit = {
```

Comment:
Loose expectation so we do not end up with 404's which lead to incoherent failures or in the case of expected 404's log output from wiremock contaminating test output. That false output can confuse when debugging failing tests, especially in a CI environment where everything is ran at once.